dashboard/app/models/ability.rb

```diff
@@ -93,6 +93,10 @@ def initialize(user)
       can :read, ScriptLevel do |script_level|
         !script_level.script.login_required?
       end
+      levels = ProjectsController::STANDALONE_PROJECTS.select { |key, value| value[:login_required] }.map { |key, value| Level.find_by_key(value[:name]).id }
+      cannot :read, Level do |level|
+        levels.include?(level.id)
+      end
```

Comment:
hmm, maybe we should make login_required a property of the level that can be set in levelbuilder?

So then this would be:
````
cannot :read, Level do |level|
  level.login_required?
end
````
and the STANDALONE_PROJECTS list doesn't have to know anything.

Or... if we want to require login for all applab levels we can just check if the level is of type applab here.